### DISCUSSION TITLE
YOSUPGO PCI PC relay tasmotized

### DISCUSSION THREAD

**Discussion by jhuettner** [arendst/Tasmota#12867] [Show and tell]
Just FYI for anyone else using this product, it is possible to install tasmota.
   Product is https://smile.amazon.com/gp/product/B084WWC9NS, manufacturer is YOSUPGO, uses the eWelink app and has an ESP8285 onboard.  Card includes headers for serial connection, nicely labeled with gnd, VCC, TX and RX.  Solder on a 4-pin header to these holes.  Connect to the GND, VCC, TX, and RX on your USB to serial adapter.  I used a SH-U09C, but any 3.3 volt adapter should work.  You do not need to reverse TX and RX.
   Make sure you know the COM port of your USB to serial adapter.  If you're not sure, run Windows device manager, expand the list of COM ports, plug it in, and note the number of the one that's added.  Then run tasmotizer with the correct COM port.
   To get the card in firmware mode, pull off the VCC power pin.  Hold down the onboard pairing button and while holding it down, plug in the VCC power again.  Green lite will start blinking.  Use tasmotizer to finish upgrading.  Works great!

**Comment by ascillato** [comment]
Hi, Thanks for sharing. Which Template or GPIO assignment did you use in Tasmota? Thanks.

**Reply by jhuettner** [reply]
The Sonoff basic works fine for turn on and turn off, at least with my multi-meter.  I tried installing in my desktop PC last night, but the system kept restarting, so I'm doing some further debugging.  May be some kind of tasmota watchguard if the wifi isn't connected?  (It wasn't fully setup)  I'll re-install in the desktop again tonight and re-test after setting the wifi.
   The card includes an ESP8286 as the primary device, and also a Nuvoton MS51FB9AE.  The specs say that's also a microcontroller.  I'm guessing (pure guess) that it is used to control the timing for how long to keep the switched output on.  There is a separate set of headers labeled "MCU Down" that I'm assuming allow connection to that chip.
   I was hoping to be able to detect the computer status (is the computer already on?) in addition to turning it on and off.  There are no other easily-available GPIO's that I can see.  The card includes some connections to the PCIe bus, it's possible one of those is wired to a GPIO that can be used to detect if the computer is on or not.  The card pulls it's 3.3 VCC off the PCIe bus' standby power.  Hopefully I can probe more when I put it back in the desktop.  I have pictures of the card if anyone needs them.

**Reply by barbudor** [reply]
There 's another similar product (not PCIe) that's use a connection to the PC speaker to determine if it's on. But I would assume that you can easily find other sources of non-standby 3.3v or 5v which can be connected to a GPIO ans configured in Tasmota as a Switch.
I wouldn't be surprised if that's not already part of the design.

If using 5v you will need a voltage divider to feed an ESP GPIO

If you have high res pictures on which we can see the PCB track, we may be able to help

**Reply by jhuettner** [reply]
Just a follow-up.  Unit works perfect with Tasmota and Sonoff basic profile.  It properly shows if the PC is on or off.  The issue I was originally having with the system constantly rebooting turned out to be miss-connection of the power switch.  There are two sets of pins.  One connects to the original power switch, the second to the motherboard.  I assumed (incorrectly) that it didn't matter which was which.  I was wrong.  The bottom set goes to the original case power switch.  The top set goes to the motherboard.  The manual shows this, but not clearly.  The card is powered by the 3.3v always-on standby power of your power supply.  It works perfectly.  Now I can shout at Alexa (or use the app remotely) or use OpenHAB to turn my PC on and off (sleep) remotely.  Works great when I want to turn it off to save power, but need to turn it on remotely for occasional RDP access.

**Comment by no1knows** [comment]
For posterity, here is the template: https://templates.blakadder.com/oisentech_SANT-SUC01.html

...and here [flash_4M.zip](https://github.com/arendst/Tasmota/files/13880817/flash_4M.zip) is a dump of the original firmware on these cards using the following command:
`esptool.exe -b 115200 --port COM3 read_flash 0x00000 0x400000 flash_4M.bin`

**Comment by imorrish** [comment]
I managed to flash from windows using https://tasmota.github.io/install/ but took lots of goes getting device into programming mode before it would write. Once it is flashed it needed a complete power off/on before initial access point and simple web ui worked.
After that it was just a matter of going into Configuration>Module and selecting Generic (18) from the dropdown, but that doesn't update the GPIO list. You have to save/reboot, then go back into module settings to update the GPIO allocation as shown in the template link above.
<img width="372" height="775" alt="image" src="https://github.com/user-attachments/assets/4cee610f-0865-440a-904e-4bbffa0d602e" />

**Reply by sfromis** [reply]
The intended approach, instead of configuring pins one by one, is to apply the template from
https://templates.blakadder.com/oisentech_SANT-SUC01.html
(See the how-to guide in the left column)

**Comment by jokerigno** [comment]
> I managed to flash from windows using https://tasmota.github.io/install/

Can you please share the link to buy the product? Link on template website no longer works and I didn't find any product so far

**Reply by imorrish** [reply]
These are the ones I got from Aliexpress. https://www.aliexpress.com/item/1005006277886826.html
